### PR TITLE
REF: ArrowEA _data->_pa_array

### PR DESCRIPTION
**Pull request by jbrockmendel**
Totally subjective: I prefer a distinctive/informative name.  This makes it easy to e.g. grep for all the places where we access this attribute.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
@mroeschke if you're not on board we can close this, otherwise ill rebase

**Comment by mroeschke** [comment]
Curious, would this make it harder for `ArrowExtensionArray` to try to use masked array methods since it calls `self._data` a lot?

**Comment by jbrockmendel** [comment]
> Curious, would this make it harder for ArrowExtensionArray to try to use masked array methods since it calls self._data a lot?

It would if we did that, but they don't share any code...

**Comment by mroeschke** [comment]
I was thinking of, for example, utilizing the base masked engine work in the short term to have better indexing support which I think utilizes `._data`.

But overall it's not that big of a deal, I am fine changing it to `_pa_array`

**Comment by jbrockmendel** [comment]
@mroeschke any objection to merging before more merge conflicts pile up?

**Comment by simonjayhawkins** [comment]
> But overall it's not that big of a deal, I am fine changing it to `_pa_array`

yep. Go ahead and change if you feel it's an improvement.

**Comment by jbrockmendel** [comment]
Just fixed another merge conflict.  im planning to merge on green unless someone objects.

**Review by simonjayhawkins — APPROVED** [review]

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel

File: pandas/core/arrays/arrow/array.py [hunk 1]
```diff
@@ -413,44 +413,44 @@ def __iter__(self) -> Iterator[Any]:
...
     # https://issues.apache.org/jira/browse/ARROW-10739 is addressed
     def __getstate__(self):
         state = self.__dict__.copy()
-        state["_data"] = self._data.combine_chunks()
+        state["_data"] = self._pa_array.combine_chunks()
         return state
 
     def __setstate__(self, state) -> None:
-        state["_data"] = pa.chunked_array(state["_data"])
+        state["_pa_array"] = pa.chunked_array(state["_data"])
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/arrays/arrow/array.py:442]: yah, looks like we'll need to check for both _data and _pa_array until older pickles are dropped

[the remark this answers is not in the document]
Comment (phofl) [reply pandas/core/arrays/arrow/array.py:442]: Do you have time to take a look?

**Pull request merged**